demos/https/iot_demo_https_sync.c

```diff
@@ -0,0 +1,499 @@
+/*
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file iot_demo_https_sync.c
+ * @brief Demonstrates synchronous usage of the HTTPS library by performing partial content GET requests on a file from S3.
+ */
+
+/* The config header is always included first. */
+#include "iot_config.h"
+
+/* C Standard includes. */
+#include <stdbool.h>
+#include <string.h>
+
+/* Set up logging for this demo. */
+#include "iot_demo_logging.h"
+
+/* Amazon FreeRTOS includes. */
+#include "iot_https_client.h"
+#include "iot_https_utils.h"
+#include "aws_demo_config.h"
+#include "platform/iot_network.h"
+#include "private/iot_error.h"
+
+/**
+ * This demo downloads an object from S3 and outputs to the console. A presigned URL is required to run this demo. Please
+ * see the demos/https/README.md for instructions on how to generate one.
+ */
+
+/**
+ * @cond DOXYGEN_IGNORE
+ * Doxygen should ignore this section.
+ *
+ * Provide default values for undefined configuration settings.
+ */
+
+/* Presigned URL for S3 GET Object access. */
+#ifndef IOT_DEMO_HTTPS_PRESIGNED_URL
+    #define IOT_DEMO_HTTPS_PRESIGNED_URL     "Please configure a presigned URL in iot_config.h."
+#endif
+
+/* TLS port for HTTPS. */
+#ifndef IOT_DEMO_HTTPS_PORT
+    #define IOT_DEMO_HTTPS_PORT              ( 443 )
+#endif
+
+#ifndef IOT_DEMO_HTTPS_TRUSTED_ROOT_CA
+    /* This the Baltomore Cybertrust issued root CA that the s3 server uses. */
```

Comment:
Reminder to update this to simply state it is is the Baltimore Cybertrust root certificate. Certificates are self-signed not issued.